src/main/webapp/js/userMap.js

```diff
@@ -104,8 +114,98 @@ function handleData(err, userData) {
             }
         }
     });
+
+    map.addPlugin('pins', function ( layer, data, options ) {
+        var self = this,
+            fillData = this.options.fills,
+            svg = this.svg;
+
+        if ( !data || (data && !data.slice) ) {
+            throw "Datamaps Error - markers must be an array";
```

Comment:
I don't think this helps because it's not handled anywhere. You can just tell the user that an error occurs, like what is done in lines 7-13.